mailpoet/lib/Services/Bridge/API.php

```diff
@@ -196,26 +220,29 @@ public function createAuthorizedEmailAddress(string $emailAddress): array {
       $body
     );
 
-    $code = $this->wp->wpRemoteRetrieveResponseCode($result);
-    $isSuccess = $code === self::RESPONSE_CODE_CREATED;
+    $responseCode = $this->wp->wpRemoteRetrieveResponseCode($result);
 
-    if (!$isSuccess) {
+    if ($responseCode !== self::RESPONSE_CODE_CREATED) {
       $errorBody = $this->wp->wpRemoteRetrieveBody($result);
       $logData = [
-        'code' => $code,
+        'code' => $responseCode,
         'error' => is_wp_error($result) ? $result->get_error_message() : $errorBody,
       ];
       $this->loggerFactory->getLogger(LoggerFactory::TOPIC_BRIDGE)->error('CreateAuthorizedEmailAddress API call failed.', $logData);
 
       $errorResponseData = json_decode($errorBody, true);
       // translators: %d is the error code.
-      $fallbackError = sprintf(__('An error has happened while performing a request, the server has responded with response code %d', 'mailpoet'), $code);
+      $fallbackError = sprintf(__('An error has happened while performing a request, the server has responded with response code %d', 'mailpoet'), $responseCode);
 
-      $errorData = is_array($errorResponseData) && isset($errorResponseData['error']) ? $errorResponseData['error'] : $fallbackError;
-      return ['error' => $errorData, 'status' => false];
+      return [
+        'status' => self::AUTHORIZED_EMAIL_STATUS_ERROR,
+        'code' => $responseCode,
+        'error' => is_array($errorResponseData) && isset($errorResponseData['error']) ? $errorResponseData['error'] : $fallbackError,
+        'message' => is_array($errorResponseData) && isset($errorResponseData['error']) ? $this->getTranslatedErrorMessage($errorResponseData['error']) : $fallbackError,
+      ];
```

Comment:
I see we construct the array of this shape in a couple of places. I really like the idea of a unified shape of the error response. 

I think we could introduce a private method (e.g. `formatErrorResponse` or some better name). This would increase the probability that methods we add in the future will use the same shape because the developer will see there is a method for formatting errors.

The translation mapping returns the original error when there is no translation so the `formatErrorResponse` method could handle adding the `message` key with translation. Also, with the common response status (see previous comment), it could have the status parameter optional defaulting to `self::RESPONSE_STATUS_ERROR`

Reply:
Good point. I will address it in a new commit. 👍